lib/config/fileconf.go

```diff
@@ -806,9 +806,6 @@ type Auth struct {
 	// This is currently Cloud-specific.
 	HostedPlugins HostedPlugins `yaml:"hosted_plugins,omitempty"`
 
-	// Assist is a set of options related to the Teleport Assist feature.
-	Assist *AuthAssistOptions `yaml:"assist,omitempty"`
-
```

Comment:
this will break `teleport` startup after upgrade if the field is set with a message like

```
ERROR: failed parsing the config file: yaml: unmarshal errors:  line 41: field assist not found in type config.Auth
```

Which I just hit trying to patch my staging tenant.

And we still document that field in both auth and proxy service config reference:
https://goteleport.com/docs/reference/config